uaclient/api/api.py

```diff
@@ -18,6 +18,7 @@
     "u.pro.attach.magic.initiate.v1",
     "u.pro.attach.magic.wait.v1",
     "u.pro.attach.magic.revoke.v1",
+    "u.pro.detect.should_auto_attach.v1",
```

Comment:
Sorry to drag the naming discussion along further, but keeping in mind the conversation here: https://github.com/canonical/ubuntu-advantage-client/pull/2171#discussion_r930243892 

I think this would fit nicely with my proposal there if it was named: `u.pro.attach.auto.should_auto_attach.v1`